src/quark.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use translate::*;
+
+#[repr(C)]
+pub struct Quark(ffi::GQuark);
+
+impl Quark {
+    pub fn from_string(s: &str) -> Quark {
+        unsafe {
+            from_glib(ffi::g_quark_from_string(s.to_glib_none().0))
```

Comment:
If you want to prevent a copy, you could make this one here use to_glib_full() and have it call the static variant of the function. Same overall behaviour but one less copy